my-app/src/components/BurgerMenu/BurgerMenu.tsx

```diff
@@ -1,15 +1,29 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { IPost } from '../Post/Post';
 import "./style.css"
 
-const BurgerMenu = () => {
+const BurgerMenu = ({ onSubmit }: any) => {
+  const [inputValue, setInputValue] = useState('');
+
+  const handleChange = (event: any) => {
+    setInputValue(event.target.value)
+  } 
+
+  onSubmit(inputValue)
```

Comment:
This function will be called EVERY TIME component gets rerendered - do we really need this? Did you mean to call once (when component is mounted)?